roles/kubernetes/master/templates/kubeadm-config.yaml.j2

```diff
@@ -16,8 +16,11 @@ networking:
   serviceSubnet: {{ kube_service_addresses }}
   podSubnet: {{ kube_pods_subnet }}
 kubernetesVersion: {{ kube_version }}
-{% if cloud_provider is defined and cloud_provider != "gce" %}
-cloudProvider: {{ cloud_provider }}
+{% if cloud_provider is defined and cloud_provider in ["openstack", "azure", "vsphere"] %}
+  cloud-provider: {{ cloud_provider }}
+  cloud-config: {{ kube_config_dir }}/cloud_config
```

Comment:
this path isn't mounted inside containers. it needs to be set as an extra vol mount. We need extra mount points for apiserver/scheduler/controller: https://github.com/kubernetes/kubernetes/pull/49840